[bug] Nomad panics when attempting to scale a job by a negative value

### Nomad version
Version: `Nomad v0.11.2-dev`
Commit: `f5c84fca8e128be8ff609f3b6bfc0230a25d3adf`

### Operating system and Environment details
Reproduced on Mac and Linux amd64.

### Issue
When attempting to scale a job task group by a negative value, the Nomad server panics.

### Reproduction steps
- run Nomad locally via `nomad agent -dev`
- register the job as defined below `nomad run <jobfile>`
- attempt to scale using the payload below `curl --request POST --data @payload.json http://localhost:4646/v1/job/example/scale`
- observe Nomad panic

### Job file
```
job "example" {
  datacenters = ["dc1"]

  group "cache" {

    count = 1

    task "redis" {
      driver = "docker"

      config {
        image = "redis:3.2"
      }
    }
  }
}

```

### API Payload
```
{
  "Count": -1,
  "Target": {
    "Group": "cache"
  }
}
```

### Nomad Logs
```
panic: bitmap must be positive size

goroutine 122 [running]:
github.com/hashicorp/nomad/scheduler.bitmapFrom(0xc001b89d40, 0xffffffffffffffff, 0x2, 0x2, 0xc001b89d40)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/reconcile_util.go:428 +0x25c
github.com/hashicorp/nomad/scheduler.newAllocNameIndex(...)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/reconcile_util.go:387
github.com/hashicorp/nomad/scheduler.(*allocReconciler).computeGroup(0xc00072b040, 0xc001396740, 0x5, 0xc001b89c20, 0xc001b89b90)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/reconcile.go:362 +0x339
github.com/hashicorp/nomad/scheduler.(*allocReconciler).Compute(0xc00072b040, 0xc001b894a0)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/reconcile.go:207 +0x170
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).computeJobAllocs(0xc00072a820, 0xc00000d2c0, 0xc0002693b0)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:347 +0x337
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).process(0xc00072a820, 0x0, 0x0, 0x0)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:254 +0x35e
github.com/hashicorp/nomad/scheduler.retryMax(0x5, 0xc000bfbe00, 0xc000bfbdf0, 0x6, 0x6f650a0)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/util.go:280 +0x56
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).Process(0xc00072a820, 0xc0013f1800, 0x6f650a0, 0xc00095eff0)
/Users/jrasell/go/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:153 +0x28c
github.com/hashicorp/nomad/nomad.(*Worker).invokeScheduler(0xc0005152d0, 0xc001b89320, 0xc0013f1800, 0xc0010cfc80, 0x24, 0x0, 0x0)
/Users/jrasell/go/src/github.com/hashicorp/nomad/nomad/worker.go:268 +0x421
github.com/hashicorp/nomad/nomad.(*Worker).run(0xc0005152d0)
/Users/jrasell/go/src/github.com/hashicorp/nomad/nomad/worker.go:129 +0x294
created by github.com/hashicorp/nomad/nomad.NewWorker
/Users/jrasell/go/src/github.com/hashicorp/nomad/nomad/worker.go:81 +0x152
```

--- a/nomad/job_endpoint.go
+++ b/nomad/job_endpoint.go
@@ -905,6 +905,9 @@ func (j *Job) Scale(args *structs.JobScaleRequest, reply *structs.JobRegisterRes
 	if args.Error && args.Count != nil {
 		return structs.NewErrRPCCoded(400, "scaling action should not contain count if error is true")
 	}
+	if args.Count != nil && *args.Count < 0 {
+		return structs.NewErrRPCCoded(400, "scaling action count can't be negative")
+	}
 
 	// Check for submit-job permissions
 	if aclObj, err := j.srv.ResolveToken(args.AuthToken); err != nil {

--- a/nomad/job_endpoint_test.go
+++ b/nomad/job_endpoint_test.go
@@ -5617,6 +5617,36 @@ func TestJobEndpoint_Scale_NoEval(t *testing.T) {
 	require.Greater(eventsIndex, jobCreateIndex)
 }
 
+func TestJobEndpoint_InvalidCount(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	s1, cleanupS1 := TestServer(t, nil)
+	defer cleanupS1()
+	codec := rpcClient(t, s1)
+	testutil.WaitForLeader(t, s1.RPC)
+	state := s1.fsm.State()
+
+	job := mock.Job()
+	err := state.UpsertJob(1000, job)
+	require.Nil(err)
+
+	scale := &structs.JobScaleRequest{
+		JobID: job.ID,
+		Target: map[string]string{
+			structs.ScalingTargetGroup: job.TaskGroups[0].Name,
+		},
+		Count: helper.Int64ToPtr(int64(-1)),
+		WriteRequest: structs.WriteRequest{
+			Region:    "global",
+			Namespace: job.Namespace,
+		},
+	}
+	var resp structs.JobRegisterResponse
+	err = msgpackrpc.CallWithCodec(codec, "Job.Scale", scale, &resp)
+	require.Error(err)
+}
+
 func TestJobEndpoint_GetScaleStatus(t *testing.T) {
 	t.Parallel()
 	require := require.New(t)

